Explorer/Assets/DCL/AvatarRendering/Emotes/Systems/SocialEmotePinsSystem.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.DefaultSystemGroups;
+using DCL.AvatarRendering.AvatarShape.Components;
+using DCL.AvatarRendering.AvatarShape.UnityInterface;
+using DCL.AvatarRendering.Emotes;
+using DCL.Character.Components;
+using DCL.CharacterCamera;
+using DCL.Diagnostics;
+using DCL.Nametags;
+using ECS.Abstract;
+using ECS.Prioritization.Components;
+using UnityEngine;
+using UnityEngine.Pool;
+
+namespace DCL.SocialEmotes.UI
+{
+    [UpdateInGroup(typeof(PreRenderingSystemGroup))]
+    [UpdateAfter(typeof(NametagPlacementSystem))]
+    [LogCategory(ReportCategory.EMOTE)]
+    public partial class SocialEmotePinsSystem : BaseUnityLoopSystem
+    {
+        private const float MAX_DISTANCE = 40;
+        private const float MAX_DISTANCE_SQR = MAX_DISTANCE * MAX_DISTANCE;
+
+        private readonly IObjectPool<SocialEmotePin> pinsPool;
+        private SingleInstanceEntity playerCamera;
+        private CameraComponent mainCameraComponent;
+        private bool isCameraSet;
+
+        public SocialEmotePinsSystem(World world, IObjectPool<SocialEmotePin> pinsPool) : base(world)
+        {
+            this.pinsPool = pinsPool;
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            playerCamera = World.CacheCamera();
+        }
+
+        protected override void Update(float t)
+        {
+            if (!isCameraSet)
+            {
+                mainCameraComponent = playerCamera.GetCameraComponent(World);
+                isCameraSet = true;
+            }
+
+            CreatePinQuery(World, mainCameraComponent);
+            UpdatePinPositionQuery(World);
+            AddNametagHeightToPinPositionQuery(World);
+            MakePinFaceCameraQuery(World, mainCameraComponent);
+            RemovePinQuery(World, mainCameraComponent);
```

Comment:
If you moved RemovePinQuery to be the first one, the other queries would avoid doing work on pins that are to be removed?